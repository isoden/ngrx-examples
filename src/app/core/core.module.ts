import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { RouterModule } from '@angular/router';

import { AppComponent } from './containers/app/app.component';
import { TopPageComponent } from './containers/top-page/top-page.component';

@NgModule({
  imports: [
    CommonModule,
    RouterModule.forChild([
      {
        path     : '',
        component: TopPageComponent,
      },
    ]),
  ],
  declarations: [
    AppComponent,
    TopPageComponent,
  ],
})
export class CoreModule { }
