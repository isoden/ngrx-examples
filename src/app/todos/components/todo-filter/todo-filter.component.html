<p>
  Filter: 
  <button (click)="onFilterChange('SHOW_ALL')"       [disabled]="filter === 'SHOW_ALL'">All</button>
  <button (click)="onFilterChange('SHOW_ACTIVE')"    [disabled]="filter === 'SHOW_ACTIVE'">Active</button>
  <button (click)="onFilterChange('SHOW_COMPLETED')" [disabled]="filter === 'SHOW_COMPLETED'">Completed</button>
</p>
