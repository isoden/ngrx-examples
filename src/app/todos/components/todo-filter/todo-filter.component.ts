import { Component, OnInit, ChangeDetectionStrategy, Output, EventEmitter, Input } from '@angular/core';
import { VisibilityFilter } from '../../models';

@Component({
  selector: 'app-todo-filter',
  templateUrl: './todo-filter.component.html',
  styleUrls: ['./todo-filter.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class TodoFilterComponent implements OnInit {
  @Input()
  filter: VisibilityFilter;

  @Output()
  filterChange = new EventEmitter<string>();

  constructor() { }

  ngOnInit() {
  }

  onFilterChange(visibilityFilter: VisibilityFilter) {
    this.filterChange.emit(visibilityFilter);
  }
}
