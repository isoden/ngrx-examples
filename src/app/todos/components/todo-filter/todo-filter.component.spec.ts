import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { ChangeDetectionStrategy } from '@angular/core';
import { By } from '@angular/platform-browser';

import { TodoFilterComponent } from './todo-filter.component';

describe('TodoFilterComponent', () => {
  let component: TodoFilterComponent;
  let fixture: ComponentFixture<TodoFilterComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [
        TodoFilterComponent,
      ],
    })
    .overrideComponent(TodoFilterComponent, {
      set: { changeDetection: ChangeDetectionStrategy.Default },
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(TodoFilterComponent);
    component = fixture.componentInstance;
    component.filter = 'SHOW_ALL';
    fixture.detectChanges();
  });

  it('should compile', () => {
    expect(fixture).toMatchSnapshot();
  });

  it('should reflect current filter', () => {
    component.filter = 'SHOW_ACTIVE';
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();

    component.filter = 'SHOW_COMPLETED';
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should emit "filterChange" on button click', () => {
    const button = fixture.debugElement.query(By.css('button:nth-of-type(2)'));
    jest.spyOn(component.filterChange, 'emit');

    button.triggerEventHandler('click', null);

    expect(component.filterChange.emit).toHaveBeenCalledWith('SHOW_ACTIVE');
  });
});
