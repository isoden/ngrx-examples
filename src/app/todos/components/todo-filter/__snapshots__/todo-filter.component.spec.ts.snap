// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoFilterComponent should compile 1`] = `
<app-todo-filter
  filter={[Function String]}
  filterChange={[Function EventEmitter]}
>
  <p>
    
    Filter: 
    
    <button
      disabled=""
    >
      All
    </button>
    
    
    <button>
      Active
    </button>
    
    
    <button>
      Completed
    </button>
    
  
  </p>
  
</app-todo-filter>
`;

exports[`TodoFilterComponent should reflect current filter 1`] = `
<app-todo-filter
  filter={[Function String]}
  filterChange={[Function EventEmitter]}
>
  <p>
    
    Filter: 
    
    <button>
      All
    </button>
    
    
    <button
      disabled=""
    >
      Active
    </button>
    
    
    <button>
      Completed
    </button>
    
  
  </p>
  
</app-todo-filter>
`;

exports[`TodoFilterComponent should reflect current filter 2`] = `
<app-todo-filter
  filter={[Function String]}
  filterChange={[Function EventEmitter]}
>
  <p>
    
    Filter: 
    
    <button>
      All
    </button>
    
    
    <button>
      Active
    </button>
    
    
    <button
      disabled=""
    >
      Completed
    </button>
    
  
  </p>
  
</app-todo-filter>
`;
